tests/cases/compiler/scopeClassDeclarations.ts

```diff
@@ -0,0 +1,19 @@
+declare const condition: boolean;
+
+if (condition)
+    class C {} // runtime error if transpiled to ES2015 or above: Unexpected token class.
+
+new C(); // runtime error if transpiled to ES5, because 'C' is undefined.
+
+if (condition) 
+    console.log(); 
+else
+    class X {}
+
+function foo() {
+	return class X {}; // Not an error.
+}
+
+while (class C {}) {  // Not an error.
+    break;
+}
```

Comment:
This file should also test the other missing parents, e.g. case, default, etc.

Reply:
Oh sure thanks!